network/p2p/peerstore/peerstore.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package peerstore
+
+import (
+	"context"
+	"fmt"
+
+	ds "github.com/ipfs/go-datastore"
+	pebbledb "github.com/ipfs/go-ds-pebble"
+	"github.com/libp2p/go-libp2p/core/peer"
+	libp2p "github.com/libp2p/go-libp2p/core/peerstore"
+	"github.com/libp2p/go-libp2p/p2p/host/peerstore/pstoreds"
+)
+
+// PeerStore implements libp2p.Peerstore
+type PeerStore struct {
+	libp2p.Peerstore
+}
+
+func initDBStore(path string) (ds.Batching, error) {
+	store, err := pebbledb.NewDatastore(path, nil)
+	return store, err
+}
+
+// NewPeerStore creates a new peerstore backed by a datastore.
+// TODO: consider using PebbleDB in the future.
+// it is currently still in experimental state. https://github.com/ipfs/go-ds-pebble
+func NewPeerStore(ctx context.Context, path string, addrInfo []*peer.AddrInfo) (*PeerStore, error) {
+	datastore, err := initDBStore(path)
+	if err != nil {
+		return nil, fmt.Errorf("cannot initialize a peerstore, invalid path for datastore: %w", err)
```

Comment:
ignoreable nit: "invalid path" seems like the most likely culprit, but maybe there are other reasons pebble could error on it's `Open()`? Also, is it safe to print out the failing path as well?